### PR TITLE
AUTO-53: Barbican - OS-2026 : Verify with secret key added to secret container - stable/5.0

### PR DESCRIPTION
**Pull request by SuhasTA4**
…

### CODE REVIEW

**Comment by SuhasTA4** [comment]
image

File: tempest/api/workloadmgr/base.py [hunk 1]
```diff
@@ -3619,6 +3620,36 @@ def delete_vm_secgroups(self, vms, secgrp_ids):
             self.delete_security_group(secgrp)
             LOG.debug("Deleted security groups: {}".format(secgrp))
 
+
+    '''
+    This method creates a secret container using secret uuid.
```
Comment (savitha-p) [inline tempest/api/workloadmgr/base.py:3625]: This method should have a cleanup call to delete the container created.

**Review by nasharatm — APPROVED** [review]
approved. Please add cleanup code as suggested by savitha.

File: tempest/api/workloadmgr/base.py (hunk 1 above)
Comment (SuhasTA4) [reply tempest/api/workloadmgr/base.py:3625]: Required changes incorporated.  Added deletion method in base.py and called in barbican test.

File: tempest/api/workloadmgr/barbican/test_barbican.py [hunk 2]
```diff
@@ -4700,3 +4700,271 @@ def test_18_barbican(self):
...
+            reporting.test_case_to_write()
+
+        except Exception as e:
+            LOG.error("Exception: " + str(e))
+            reporting.add_test_step(str(e), tvaultconf.FAIL)
+            reporting.set_test_script_status(tvaultconf.FAIL)
+
+        finally:
+            #delete secret container created.
+            self.delete_secret_container(self.container_ref)
```
Comment (savitha-p) [inline tempest/api/workloadmgr/barbican/test_barbican.py]: This need to be added to base.py as default cleanup call and not to be added here explicitly. You can refer to create_secret() method and have similar implementation.

**Review by savitha-p — APPROVED** [review]

**Pull request merged**